metricbeat/module/kibana/stats/data_xpack.go

```diff
@@ -0,0 +1,195 @@
+				"total": c.Int("total"),
+			}),
+			"search": c.Dict("kibana.search", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"visualization": c.Dict("kibana.visualization", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"dashboard": c.Dict("kibana.dashboard", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"timelion_sheet": c.Dict("kibana.timelion_sheet", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"graph_workspace": c.Dict("kibana.graph_workspace", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"xpack": s.Object{
+				"reporting": c.Dict("reporting", s.Schema{
+					"available":     c.Bool("available"),
+					"enabled":       c.Bool("enabled"),
+					"browser_type":  c.Str("browser_type"),
+					"_all":          c.Int("all"),
+					"csv":           reportingCsvDict,
+					"printable_pdf": reportingPrintablePdfDict,
+					"status":        reportingStatusDict,
+					"lastDay":       c.Dict("last_day", reportingPeriodSchema, c.DictOptional),
+					"last7Days":     c.Dict("last_7_days", reportingPeriodSchema, c.DictOptional),
+				}, c.DictOptional),
+			},
+		}),
+	}
+
+	reportingCsvDict = c.Dict("csv", s.Schema{
+		"available": c.Bool("available"),
+		"total":     c.Int("total"),
+	}, c.DictOptional)
+
+	reportingPrintablePdfDict = c.Dict("printable_pdf", s.Schema{
+		"available": c.Bool("available"),
+		"total":     c.Int("total"),
+		"app": c.Dict("app", s.Schema{
+			"visualization": c.Int("visualization"),
+			"dashboard":     c.Int("dashboard"),
+		}, c.DictOptional),
+		"layout": c.Dict("layout", s.Schema{
+			"print":           c.Int("print"),
+			"preserve_layout": c.Int("preserve_layout"),
+		}, c.DictOptional),
+	}, c.DictOptional)
+
+	reportingStatusDict = c.Dict("status", s.Schema{
+		"completed":  c.Int("completed", s.Optional),
+		"failed":     c.Int("failed", s.Optional),
+		"processing": c.Int("processing", s.Optional),
+		"pending":    c.Int("pending", s.Optional),
+	}, c.DictOptional)
+
+	reportingPeriodSchema = s.Schema{
+		"_all":          c.Int("all"),
+		"csv":           reportingCsvDict,
+		"printable_pdf": reportingPrintablePdfDict,
+		"status":        reportingStatusDict,
+	}
+)
+
+func reportErrorForMissingField(field string, r mb.ReporterV2) error {
+	err := fmt.Errorf("Could not find field '%v' in Kibana stats API response", field)
+	r.Error(err)
+	return err
+}
+
+func eventMappingXPack(r mb.ReporterV2, intervalMs int64, content []byte) error {
+	var data map[string]interface{}
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	kibanaStatsFields, err := schemaXPackMonitoring.Apply(data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	process, ok := data["process"].(map[string]interface{})
+	if !ok {
+		return reportErrorForMissingField("process", r)
+	}
+	memory, ok := process["memory"].(map[string]interface{})
+	if !ok {
+		return reportErrorForMissingField("process.memory", r)
+	}
+
+	rss, ok := memory["resident_set_size_bytes"].(float64)
```

Comment:
why do you use `float64`  here?

Reply:
Originally I tried to use `int` here but the type assertion failed. The value in the JSON response from the Kibana stats API looks like an integer to me but for some reason the [JSON parser thinks its a float because perhaps it's too large](https://stackoverflow.com/questions/22343083/json-marshaling-with-long-numbers-in-golang-gives-floating-point-number)?.

Reply:
Perhaps you should try `int64`? But I would be surprised if the value exceeds int TBH.

Reply:
Yeah, it didn't like `int64` either :(.

FWIW, if I do a `fmt.Println(memory["resident_set_size_bytes"])` (right before the type assertion), I get `6.2808064e+07`.

Reply:
Ah, I see. Kibana does report it in this format. float and later converstion to int SGTM. Perhaps make it int64 there.

Reply:
Done in b3fbd5f.